pkg/ctl/utils/update_kube_proxy.go

```diff
@@ -75,5 +75,12 @@ func doUpdateKubeProxy(p *api.ProviderConfig, cfg *api.ClusterConfig, nameArg st
 		return err
 	}
 
-	return defaultaddons.UpdateKubeProxyImageTag(clientSet, cprv, false)
+	updateRequired, err := defaultaddons.UpdateKubeProxyImageTag(clientSet, cprv, plan)
+	if err != nil {
+		return err
+	}
+
+	cmdutils.LogPlanModeWarning(plan && updateRequired)
```

Comment:
I think we would get cleaner APIs if we remove the return boolean from `UpdateKubeProxyImageTag()`, `UpdateCoreDNSImageTag()`, etc. and instead, just move the call to `cmdutils.LogPlanModeWarning()` inside those functions instead

Reply:
Yes, but see my comment above... And more specifically - we should avoid calling any `cmdutils` function from a library function. That package is for CLI contexts only.